src/completion_queue.rs

```diff
@@ -40,7 +42,10 @@ impl CompletionQueue {
                 0,
             )
         })
-        .ok_or(io::ErrorKind::Other)?;
+        // On failure, errno indicates the failure reason:
+        // EINVAL	Invalid cqe, channel or comp_vector
+        // ENOMEM	Not enough resources to complete this operation
```

Comment:
Shall we also put the error return value documents on the top of the function?

Reply:
It's a little redundant, but maybe more user-friendly.
I think the better way is match these errno for every ibv API and put related msg in return value like how we deal with `ibv_post_send`.
But considering that these APIs are not as error-prone and complex as `ibv_post_send`, so I didn't do it, maybe later : )

Reply:
I put error doc on the top of functions and mark the relevant ibv API's name, does this look better?